src/cli.rs

```diff
@@ -1,161 +1,114 @@
-                .requires("Password")
-                .help("Username to access the mqtt broker")
-                .long_help(
-                    "Username to access the mqtt broker. Anonymous access when not supplied.",
-                ),
-        )
-        .arg(
-            Arg::new("Password")
-                .long("password")
-                .env("MQTTUI_PASSWORD")
-                .value_hint(ValueHint::Other)
-                .value_name("STRING")
-                .global(true)
-                .hide_env_values(true)
-                .takes_value(true)
-                .requires("Username")
-                .help("Password to access the mqtt broker")
-                .long_help(
-                    "Password to access the mqtt broker. Passing the password via command line is insecure as the password can be read from the history!",
-                ),
-        )
-        .arg(
-            Arg::new("Topic")
-                .env("MQTTUI_TOPIC")
-                .value_hint(ValueHint::Other)
-                .value_name("TOPIC")
-                .takes_value(true)
-                .default_value("#")
-                .help("Topic to watch"),
-        )
+#[derive(Parser, Debug)]
+pub enum SubCommands {
+    /// Clean retained messages from the broker
+    ///
+    /// Clean retained messages from the broker. This works by subscribing to the topic
+    /// and waiting for messages with the retained flag. Then a message with an empty payload
+    /// is published retained which clears  the topic on the broker. Ends on the
+    /// first non retained message or when the timeout is reached
+    #[clap(visible_alias = "c", visible_alias = "clean")]
+    CleanRetained {
+        /// Topic which gets cleaned, Supports filters like 'foo/bar/#'
+        #[clap(required = true)]
+        topic: String,
+        /// When there is no message received for the given time the operation is considered done
+        #[clap(long, default_value = "5")]
+        timeout: f32,
+        // Dont clean topics, only log them
+        #[clap(long)]
+        dry_run: bool,
+    },
+    /// Log values from subscribed topics to stdout
+    #[clap(visible_alias = "l")]
+    Log {
+        /// Topics to log
+        #[clap(required = true, env = "MQTT_TOPIC", default_value = "#")]
+        topic: Vec<String>,
+
+        /// Show full MQTT communication
+        #[clap(short, long)]
+        verbose: bool,
+    },
+
+    #[clap(visible_alias = "p", visible_alias = "pub")]
+    Publish {
+        /// Topic to publish to,
+        #[clap(required = true)]
+        topic: String,
+
+        /// Payload to be published
+        #[clap(required = true, name = "PAYLOAD")]
+        payload: String,
+
+        /// Publish the MQTT message retained
+        #[clap(short, long)]
+        retain: bool,
+
+        /// Show full MQTT communication
+        #[clap(short, long)]
+        verbose: bool,
+    },
+}
+
+#[derive(Parser, Debug)]
+pub struct Cli {
```

Comment:
Not sure what's also missing in there but at least its required to have the same `--help` like before

```suggestion
#[clap(about, author, version)
pub struct Cli {
```